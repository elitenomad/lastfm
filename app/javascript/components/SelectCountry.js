import React from "react";
import PropTypes from "prop-types";
import Autocomplete  from 'react-autocomplete';

const COUNTRIES = require('../utils/countries');

class SelectCountry extends React.Component {

    render(){
        const { selectedCountry, onSelect, onFilter } = this.props;
        return(
            <div className='countries'>
                <Autocomplete
                    getItemValue={(item) => item.label}
                    items={ COUNTRIES.countries }
                    renderItem={(item, isHighlighted) =>
                        <div style={{ background: isHighlighted ? 'lightgray' : 'white' }} key={item.label}>
                            {item.label}
                        </div>
                    }
                    value={selectedCountry}
                    onChange={(e) => onSelect(e.target.value) }
                    onSelect={(val) => onSelect(val)}
                />

            </div>
        )
    }
}


SelectCountry.propTypes = {
    selectedCountry: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
};

export default SelectCountry;