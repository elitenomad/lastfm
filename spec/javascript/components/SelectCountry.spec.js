import SelectCountry from './../../../app/javascript/components/SelectCountry';
import React from 'react';
import { shallow } from 'enzyme';

describe('SelectCountry Component', () => {
    it('is expected to render correctly', () => {
        const onChange = jest.fn();
        const value = 'Australia';
        const wrapper = shallow(<SelectCountry  selectedCountry='spain'  onSelect={onChange} />);
        expect(wrapper).toMatchSnapshot();
        wrapper.find('Autocomplete').simulate('change', {
            target: { value },
        });
        expect(onChange).toBeCalledWith(value);
    });
});


