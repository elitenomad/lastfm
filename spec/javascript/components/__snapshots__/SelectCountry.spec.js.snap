// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCountry Component is expected to render correctly 1`] = `
<div
  className="countries"
>
  <Autocomplete
    autoHighlight={true}
    getItemValue={[Function]}
    inputProps={Object {}}
    isItemSelectable={[Function]}
    items={
      Array [
        Object {
          "label": "Select Country",
          "value": "selected",
        },
        Object {
          "label": "United States",
          "value": "United States",
        },
        Object {
          "label": "United Kingdom",
          "value": "United Kingdom",
        },
        Object {
          "label": "Afghanistan",
          "value": "Afghanistan",
        },
        Object {
          "label": "Albania",
          "value": "Albania",
        },
        Object {
          "label": "Algeria",
          "value": "Algeria",
        },
        Object {
          "label": "American Samoa",
          "value": "American Samoa",
        },
        Object {
          "label": "Andorra",
          "value": "Andorra",
        },
        Object {
          "label": "Angola",
          "value": "Angola",
        },
        Object {
          "label": "Anguilla",
          "value": "Anguilla",
        },
        Object {
          "label": "Antarctica",
          "value": "Antarctica",
        },
        Object {
          "label": "Antigua and Barbuda",
          "value": "Antigua and Barbuda",
        },
        Object {
          "label": "Argentina",
          "value": "Argentina",
        },
        Object {
          "label": "Armenia",
          "value": "Armenia",
        },
        Object {
          "label": "Aruba",
          "value": "Aruba",
        },
        Object {
          "label": "Australia",
          "value": "Australia",
        },
        Object {
          "label": "Austria",
          "value": "Austria",
        },
        Object {
          "label": "Azerbaijan",
          "value": "Azerbaijan",
        },
        Object {
          "label": "Bahamas",
          "value": "Bahamas",
        },
        Object {
          "label": "Bahrain",
          "value": "Bahrain",
        },
        Object {
          "label": "Bangladesh",
          "value": "Bangladesh",
        },
        Object {
          "label": "Barbados",
          "value": "Barbados",
        },
        Object {
          "label": "Belarus",
          "value": "Belarus",
        },
        Object {
          "label": "Belgium",
          "value": "Belgium",
        },
        Object {
          "label": "Belize",
          "value": "Belize",
        },
        Object {
          "label": "Benin",
          "value": "Benin",
        },
        Object {
          "label": "Bermuda",
          "value": "Bermuda",
        },
        Object {
          "label": "Bhutan",
          "value": "Bhutan",
        },
        Object {
          "label": "Bolivia",
          "value": "Bolivia",
        },
        Object {
          "label": "Bosnia and Herzegovina",
          "value": "Bosnia and Herzegovina",
        },
        Object {
          "label": "Botswana",
          "value": "Botswana",
        },
        Object {
          "label": "Bouvet Island",
          "value": "Bouvet Island",
        },
        Object {
          "label": "Brazil",
          "value": "Brazil",
        },
        Object {
          "label": "British Indian Ocean Territory",
          "value": "British Indian Ocean Territory",
        },
        Object {
          "label": "Brunei Darussalam",
          "value": "Brunei Darussalam",
        },
        Object {
          "label": "Bulgaria",
          "value": "Bulgaria",
        },
        Object {
          "label": "Burkina Faso",
          "value": "Burkina Faso",
        },
        Object {
          "label": "Burundi",
          "value": "Burundi",
        },
        Object {
          "label": "Cambodia",
          "value": "Cambodia",
        },
        Object {
          "label": "Cameroon",
          "value": "Cameroon",
        },
        Object {
          "label": "Canada",
          "value": "Canada",
        },
        Object {
          "label": "Cape Verde",
          "value": "Cape Verde",
        },
        Object {
          "label": "Cayman Islands",
          "value": "Cayman Islands",
        },
        Object {
          "label": "Central African Republic",
          "value": "Central African Republic",
        },
        Object {
          "label": "Chad",
          "value": "Chad",
        },
        Object {
          "label": "Chile",
          "value": "Chile",
        },
        Object {
          "label": "China",
          "value": "China",
        },
        Object {
          "label": "Christmas Island",
          "value": "Christmas Island",
        },
        Object {
          "label": "Cocos (Keeling) Islands",
          "value": "Cocos (Keeling) Islands",
        },
        Object {
          "label": "Colombia",
          "value": "Colombia",
        },
        Object {
          "label": "Comoros",
          "value": "Comoros",
        },
        Object {
          "label": "Congo",
          "value": "Congo",
        },
        Object {
          "label": "Congo, The Democratic Republic of The",
          "value": "Congo, The Democratic Republic of The",
        },
        Object {
          "label": "Cook Islands",
          "value": "Cook Islands",
        },
        Object {
          "label": "Costa Rica",
          "value": "Costa Rica",
        },
        Object {
          "label": "Cote D'ivoire",
          "value": "Cote D'ivoire",
        },
        Object {
          "label": "Croatia",
          "value": "Croatia",
        },
        Object {
          "label": "Cuba",
          "value": "Cuba",
        },
        Object {
          "label": "Cyprus",
          "value": "Cyprus",
        },
        Object {
          "label": "Czech Republic",
          "value": "Czech Republic",
        },
        Object {
          "label": "Denmark",
          "value": "Denmark",
        },
        Object {
          "label": "Djibouti",
          "value": "Djibouti",
        },
        Object {
          "label": "Dominica",
          "value": "Dominica",
        },
        Object {
          "label": "Dominican Republic",
          "value": "Dominican Republic",
        },
        Object {
          "label": "Ecuador",
          "value": "Ecuador",
        },
        Object {
          "label": "Egypt",
          "value": "Egypt",
        },
        Object {
          "label": "El Salvador",
          "value": "El Salvador",
        },
        Object {
          "label": "Equatorial Guinea",
          "value": "Equatorial Guinea",
        },
        Object {
          "label": "Eritrea",
          "value": "Eritrea",
        },
        Object {
          "label": "Estonia",
          "value": "Estonia",
        },
        Object {
          "label": "Ethiopia",
          "value": "Ethiopia",
        },
        Object {
          "label": "Falkland Islands (Malvinas)",
          "value": "Falkland Islands (Malvinas)",
        },
        Object {
          "label": "Faroe Islands",
          "value": "Faroe Islands",
        },
        Object {
          "label": "Fiji",
          "value": "Fiji",
        },
        Object {
          "label": "Finland",
          "value": "Finland",
        },
        Object {
          "label": "France",
          "value": "France",
        },
        Object {
          "label": "French Guiana",
          "value": "French Guiana",
        },
        Object {
          "label": "French Polynesia",
          "value": "French Polynesia",
        },
        Object {
          "label": "French Southern Territories",
          "value": "French Southern Territories",
        },
        Object {
          "label": "Gabon",
          "value": "Gabon",
        },
        Object {
          "label": "Gambia",
          "value": "Gambia",
        },
        Object {
          "label": "Georgia",
          "value": "Georgia",
        },
        Object {
          "label": "Germany",
          "value": "Germany",
        },
        Object {
          "label": "Ghana",
          "value": "Ghana",
        },
        Object {
          "label": "Gibraltar",
          "value": "Gibraltar",
        },
        Object {
          "label": "Greece",
          "value": "Greece",
        },
        Object {
          "label": "Greenland",
          "value": "Greenland",
        },
        Object {
          "label": "Grenada",
          "value": "Grenada",
        },
        Object {
          "label": "Guadeloupe",
          "value": "Guadeloupe",
        },
        Object {
          "label": "Guam",
          "value": "Guam",
        },
        Object {
          "label": "Guatemala",
          "value": "Guatemala",
        },
        Object {
          "label": "Guinea",
          "value": "Guinea",
        },
        Object {
          "label": "Guinea-bissau",
          "value": "Guinea-bissau",
        },
        Object {
          "label": "Guyana",
          "value": "Guyana",
        },
        Object {
          "label": "Haiti",
          "value": "Haiti",
        },
        Object {
          "label": "Heard Island and Mcdonald Islands",
          "value": "Heard Island and Mcdonald Islands",
        },
        Object {
          "label": "Holy See (Vatican City State)",
          "value": "Holy See (Vatican City State)",
        },
        Object {
          "label": "Honduras",
          "value": "Honduras",
        },
        Object {
          "label": "Hong Kong",
          "value": "Hong Kong",
        },
        Object {
          "label": "Hungary",
          "value": "Hungary",
        },
        Object {
          "label": "Iceland",
          "value": "Iceland",
        },
        Object {
          "label": "India",
          "value": "India",
        },
        Object {
          "label": "Indonesia",
          "value": "Indonesia",
        },
        Object {
          "label": "Iran, Islamic Republic of",
          "value": "Iran, Islamic Republic of",
        },
        Object {
          "label": "Iraq",
          "value": "Iraq",
        },
        Object {
          "label": "Ireland",
          "value": "Ireland",
        },
        Object {
          "label": "Israel",
          "value": "Israel",
        },
        Object {
          "label": "Italy",
          "value": "Italy",
        },
        Object {
          "label": "Jamaica",
          "value": "Jamaica",
        },
        Object {
          "label": "Japan",
          "value": "Japan",
        },
        Object {
          "label": "Jordan",
          "value": "Jordan",
        },
        Object {
          "label": "Kazakhstan",
          "value": "Kazakhstan",
        },
        Object {
          "label": "Kenya",
          "value": "Kenya",
        },
        Object {
          "label": "Kiribati",
          "value": "Kiribati",
        },
        Object {
          "label": "Korea, Democratic People's Republic of",
          "value": "Korea, Democratic People's Republic of",
        },
        Object {
          "label": "Korea, Republic of",
          "value": "Korea, Republic of",
        },
        Object {
          "label": "Kuwait",
          "value": "Kuwait",
        },
        Object {
          "label": "Kyrgyzstan",
          "value": "Kyrgyzstan",
        },
        Object {
          "label": "Lao People's Democratic Republic",
          "value": "Lao People's Democratic Republic",
        },
        Object {
          "label": "Latvia",
          "value": "Latvia",
        },
        Object {
          "label": "Lebanon",
          "value": "Lebanon",
        },
        Object {
          "label": "Lesotho",
          "value": "Lesotho",
        },
        Object {
          "label": "Liberia",
          "value": "Liberia",
        },
        Object {
          "label": "Libyan Arab Jamahiriya",
          "value": "Libyan Arab Jamahiriya",
        },
        Object {
          "label": "Liechtenstein",
          "value": "Liechtenstein",
        },
        Object {
          "label": "Lithuania",
          "value": "Lithuania",
        },
        Object {
          "label": "Luxembourg",
          "value": "Luxembourg",
        },
        Object {
          "label": "Macao",
          "value": "Macao",
        },
        Object {
          "label": "Macedonia, The Former Yugoslav Republic of",
          "value": "Macedonia, The Former Yugoslav Republic of",
        },
        Object {
          "label": "Madagascar",
          "value": "Madagascar",
        },
        Object {
          "label": "Malawi",
          "value": "Malawi",
        },
        Object {
          "label": "Malaysia",
          "value": "Malaysia",
        },
        Object {
          "label": "Maldives",
          "value": "Maldives",
        },
        Object {
          "label": "Mali",
          "value": "Mali",
        },
        Object {
          "label": "Malta",
          "value": "Malta",
        },
        Object {
          "label": "Marshall Islands",
          "value": "Marshall Islands",
        },
        Object {
          "label": "Martinique",
          "value": "Martinique",
        },
        Object {
          "label": "Mauritania",
          "value": "Mauritania",
        },
        Object {
          "label": "Mauritius",
          "value": "Mauritius",
        },
        Object {
          "label": "Mayotte",
          "value": "Mayotte",
        },
        Object {
          "label": "Mexico",
          "value": "Mexico",
        },
        Object {
          "label": "Micronesia, Federated States of",
          "value": "Micronesia, Federated States of",
        },
        Object {
          "label": "Moldova, Republic of",
          "value": "Moldova, Republic of",
        },
        Object {
          "label": "Monaco",
          "value": "Monaco",
        },
        Object {
          "label": "Mongolia",
          "value": "Mongolia",
        },
        Object {
          "label": "Montserrat",
          "value": "Montserrat",
        },
        Object {
          "label": "Morocco",
          "value": "Morocco",
        },
        Object {
          "label": "Mozambique",
          "value": "Mozambique",
        },
        Object {
          "label": "Myanmar",
          "value": "Myanmar",
        },
        Object {
          "label": "Namibia",
          "value": "Namibia",
        },
        Object {
          "label": "Nauru",
          "value": "Nauru",
        },
        Object {
          "label": "Nepal",
          "value": "Nepal",
        },
        Object {
          "label": "Netherlands",
          "value": "Netherlands",
        },
        Object {
          "label": "Netherlands Antilles",
          "value": "Netherlands Antilles",
        },
        Object {
          "label": "New Caledonia",
          "value": "New Caledonia",
        },
        Object {
          "label": "New Zealand",
          "value": "New Zealand",
        },
        Object {
          "label": "Nicaragua",
          "value": "Nicaragua",
        },
        Object {
          "label": "Niger",
          "value": "Niger",
        },
        Object {
          "label": "Nigeria",
          "value": "Nigeria",
        },
        Object {
          "label": "Niue",
          "value": "Niue",
        },
        Object {
          "label": "Norfolk Island",
          "value": "Norfolk Island",
        },
        Object {
          "label": "Northern Mariana Islands",
          "value": "Northern Mariana Islands",
        },
        Object {
          "label": "Norway",
          "value": "Norway",
        },
        Object {
          "label": "Oman",
          "value": "Oman",
        },
        Object {
          "label": "Pakistan",
          "value": "Pakistan",
        },
        Object {
          "label": "Palau",
          "value": "Palau",
        },
        Object {
          "label": "Palestinian Territory, Occupied",
          "value": "Palestinian Territory, Occupied",
        },
        Object {
          "label": "Panama",
          "value": "Panama",
        },
        Object {
          "label": "Papua New Guinea",
          "value": "Papua New Guinea",
        },
        Object {
          "label": "Paraguay",
          "value": "Paraguay",
        },
        Object {
          "label": "Peru",
          "value": "Peru",
        },
        Object {
          "label": "Philippines",
          "value": "Philippines",
        },
        Object {
          "label": "Pitcairn",
          "value": "Pitcairn",
        },
        Object {
          "label": "Poland",
          "value": "Poland",
        },
        Object {
          "label": "Portugal",
          "value": "Portugal",
        },
        Object {
          "label": "Puerto Rico",
          "value": "Puerto Rico",
        },
        Object {
          "label": "Qatar",
          "value": "Qatar",
        },
        Object {
          "label": "Reunion",
          "value": "Reunion",
        },
        Object {
          "label": "Romania",
          "value": "Romania",
        },
        Object {
          "label": "Russian Federation",
          "value": "Russian Federation",
        },
        Object {
          "label": "Rwanda",
          "value": "Rwanda",
        },
        Object {
          "label": "Saint Helena",
          "value": "Saint Helena",
        },
        Object {
          "label": "Saint Kitts and Nevis",
          "value": "Saint Kitts and Nevis",
        },
        Object {
          "label": "Saint Lucia",
          "value": "Saint Lucia",
        },
        Object {
          "label": "Saint Pierre and Miquelon",
          "value": "Saint Pierre and Miquelon",
        },
        Object {
          "label": "Saint Vincent and The Grenadines",
          "value": "Saint Vincent and The Grenadines",
        },
        Object {
          "label": "Samoa",
          "value": "Samoa",
        },
        Object {
          "label": "San Marino",
          "value": "San Marino",
        },
        Object {
          "label": "Sao Tome and Principe",
          "value": "Sao Tome and Principe",
        },
        Object {
          "label": "Saudi Arabia",
          "value": "Saudi Arabia",
        },
        Object {
          "label": "Senegal",
          "value": "Senegal",
        },
        Object {
          "label": "Serbia and Montenegro",
          "value": "Serbia and Montenegro",
        },
        Object {
          "label": "Seychelles",
          "value": "Seychelles",
        },
        Object {
          "label": "Sierra Leone",
          "value": "Sierra Leone",
        },
        Object {
          "label": "Singapore",
          "value": "Singapore",
        },
        Object {
          "label": "Slovakia",
          "value": "Slovakia",
        },
        Object {
          "label": "Slovenia",
          "value": "Slovenia",
        },
        Object {
          "label": "Solomon Islands",
          "value": "Solomon Islands",
        },
        Object {
          "label": "Somalia",
          "value": "Somalia",
        },
        Object {
          "label": "South Africa",
          "value": "South Africa",
        },
        Object {
          "label": "South Georgia and The South Sandwich Islands",
          "value": "South Georgia and The South Sandwich Islands",
        },
        Object {
          "label": "Spain",
          "value": "Spain",
        },
        Object {
          "label": "Sri Lanka",
          "value": "Sri Lanka",
        },
        Object {
          "label": "Sudan",
          "value": "Sudan",
        },
        Object {
          "label": "Suriname",
          "value": "Suriname",
        },
        Object {
          "label": "Svalbard and Jan Mayen",
          "value": "Svalbard and Jan Mayen",
        },
        Object {
          "label": "Swaziland",
          "value": "Swaziland",
        },
        Object {
          "label": "Sweden",
          "value": "Sweden",
        },
        Object {
          "label": "Switzerland",
          "value": "Switzerland",
        },
        Object {
          "label": "Syrian Arab Republic",
          "value": "Syrian Arab Republic",
        },
        Object {
          "label": "Taiwan, Province of China",
          "value": "Taiwan, Province of China",
        },
        Object {
          "label": "Tajikistan",
          "value": "Tajikistan",
        },
        Object {
          "label": "Tanzania, United Republic of",
          "value": "Tanzania, United Republic of",
        },
        Object {
          "label": "Thailand",
          "value": "Thailand",
        },
        Object {
          "label": "Timor-leste",
          "value": "Timor-leste",
        },
        Object {
          "label": "Togo",
          "value": "Togo",
        },
        Object {
          "label": "Tokelau",
          "value": "Tokelau",
        },
        Object {
          "label": "Tonga",
          "value": "Tonga",
        },
        Object {
          "label": "Trinidad and Tobago",
          "value": "Trinidad and Tobago",
        },
        Object {
          "label": "Tunisia",
          "value": "Tunisia",
        },
        Object {
          "label": "Turkey",
          "value": "Turkey",
        },
        Object {
          "label": "Turkmenistan",
          "value": "Turkmenistan",
        },
        Object {
          "label": "Turks and Caicos Islands",
          "value": "Turks and Caicos Islands",
        },
        Object {
          "label": "Tuvalu",
          "value": "Tuvalu",
        },
        Object {
          "label": "Uganda",
          "value": "Uganda",
        },
        Object {
          "label": "Ukraine",
          "value": "Ukraine",
        },
        Object {
          "label": "United Arab Emirates",
          "value": "United Arab Emirates",
        },
        Object {
          "label": "United States Minor Outlying Islands",
          "value": "United States Minor Outlying Islands",
        },
        Object {
          "label": "Uruguay",
          "value": "Uruguay",
        },
        Object {
          "label": "Uzbekistan",
          "value": "Uzbekistan",
        },
        Object {
          "label": "Vanuatu",
          "value": "Vanuatu",
        },
        Object {
          "label": "Venezuela",
          "value": "Venezuela",
        },
        Object {
          "label": "Viet Nam",
          "value": "Viet Nam",
        },
        Object {
          "label": "Virgin Islands, British",
          "value": "Virgin Islands, British",
        },
        Object {
          "label": "Virgin Islands, U.S.",
          "value": "Virgin Islands, U.S.",
        },
        Object {
          "label": "Wallis and Futuna",
          "value": "Wallis and Futuna",
        },
        Object {
          "label": "Western Sahara",
          "value": "Western Sahara",
        },
        Object {
          "label": "Yemen",
          "value": "Yemen",
        },
        Object {
          "label": "Zambia",
          "value": "Zambia",
        },
        Object {
          "label": "Zimbabwe",
          "value": "Zimbabwe",
        },
      ]
    }
    menuStyle={
      Object {
        "background": "rgba(255, 255, 255, 0.9)",
        "borderRadius": "3px",
        "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
        "fontSize": "90%",
        "maxHeight": "50%",
        "overflow": "auto",
        "padding": "2px 0",
        "position": "fixed",
      }
    }
    onChange={[Function]}
    onMenuVisibilityChange={[Function]}
    onSelect={[Function]}
    renderInput={[Function]}
    renderItem={[Function]}
    renderMenu={[Function]}
    selectOnBlur={false}
    value="spain"
    wrapperProps={Object {}}
    wrapperStyle={
      Object {
        "display": "inline-block",
      }
    }
  />
</div>
`;
